contracts/DandelionOrg.sol

```diff
@@ -61,15 +61,13 @@ contract DandelionOrg is BaseTemplate {
     *      to be setup due to gas limits.
     * @param _tokenName String with the name for the token used by share holders in the organization
     * @param _tokenSymbol String with the symbol for the token used by share holders in the organization
-    * @param _id String with the name for org, will assign `[id].aragonid.eth`
     * @param _holders Array of token holder addresses
     * @param _stakes Array of token stakes for holders (token has 18 decimals, multiply token amount `* 10^18`)
     * @param _useAgentAsVault Boolean to tell whether to use an Agent app as a more advanced form of Vault app
     */
     function newTokenAndBaseInstance(
         string _tokenName,
         string _tokenSymbol,
-        string _id,
```

Comment:
So we have two external functions that are called the newTokenAndBaseInstance and the installDandelionApps, from the onboarding website first is called the newTokenAndBaseInstance and after the installDandelionApps. Shouldn't we validate the _id on the first call? (newTokenAndBaseInstance) store that id, remove the _id Parameter from the installDandelionApps and get that cached id from there?

Reply:
Is it necessary to have it that way round to satisfy the interface of the onboarding website? Also would the UX of the onboarding website change at all having it the way round it is currently?

Reply:
No, we can leave on the installDandelionApps as you did, but my question is if that id validation fails isn't better to fail on the first trx call (newTokenAndBaseInstance)  instead of calling newTokenAndBaseInstance and then failing on the installDandelionApps ?

Not sure about the ux question, i think that should't change

Reply:
From the onboarding website the id validation is made in the frontend before continuing with configuring the DAO, so probably the id passed to `installDandelionApps` will be a valid one except for edge cases. 

I think it makes sense to do as @rperez89 suggested. The DAO id should be "reserved" whenever the DAO is created and that happens in the first tx. If we have two different users creating a dandelion Org with the same id at the same time, the first tx will go through for both but the second one will fail for one of them and from the UX perspective of the onboarding, this user will have to create a new DAO.

Reply:
After having a look at the code, it seems we don't register the ID until the second transaction, so as it stands, if the ID is cached in the first transaction and used in the second and 2 DAO's with the same ID are created at the same time, one of them will execute both transactions while the other will be prevented from creating the second transaction (as it uses a cached ID that is already registered). As opposed to the current approach where the late user will be able to execute the second transaction, they will just have to use a different ID.

Alternatively we can register the ID in the first transaction, which I'm in favour of. We won't need it at all in the second transaction then. So do you know of any reason why we shouldn't register the ID in the first transaction?

Reply:
I think that as we have 2 transactions if we register the dao id in the fist one and the second transaction fails the user will not be able to use the same id when he try again right?

Reply:
Also a good point, they will not be able to use the same ID to create a different DAO. However, if the second transaction fails for one user using this template, it should fail for everyone using this template/every other time they try to create a DAO using it.

Reply:
Had a chat with @rperez89 and decided we should revert to the original approach, accepting the ID in the second transaction. Apparently if the second transaction fails and the user leaves and returns to the onboarding website, they won't have the chance to execute just the second transaction and will have to create a new ID to be able to submit a new first transaction, which I believe is more likely than the edge case where two people attempt to use the same ID at the same time. @fabriziovigevani Any further thoughts?

Reply:
Yeah that's a valid point, better allow to reuse an ID in case of any issues and agree that it is more likely for the user to close the site on a failing tx than two people registering same id at the same time.

Besides we would have the same problem of the user having to create a new DAO either way.